integration/src/wallets/metamask.ts

```diff
@@ -0,0 +1,143 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as metamask from "@shapeshiftoss/hdwallet-metamask";
+
+export function name(): string {
+  return "MetaMask";
+}
+
+export function createInfo(): core.HDWalletInfo {
+  return metamask.info();
+}
+
+export async function createWallet(): Promise<core.HDWallet> {
+  const wallet = new metamask.MetaMaskHDWallet();
+  await wallet.initialize();
+  return wallet;
+}
+
+export function selfTest(get: () => core.HDWallet): void {
+  let wallet: metamask.MetaMaskHDWallet;
+
+  beforeAll(async () => {
+    let w = get() as metamask.MetaMaskHDWallet;
+
+    if (metamask.isMetaMask(w) && !core.supportsBTC(w) && core.supportsETH(w)) {
+      wallet = w;
+    } else {
+      fail("Wallet is not a MetaMask");
+    }
+  });
+
+  it("supports Ethereum mainnet", async () => {
+    if (!wallet) return;
+    expect(await wallet.ethSupportsNetwork()).toEqual(true);
+  });
+
+  it("does not support BTC", async () => {
+    if (!wallet) return;
+    expect(core.supportsBTC(wallet)).toBe(false);
+  });
+
+  it("does not support Native ShapeShift", async () => {
+    if (!wallet) return;
+    expect(wallet.ethSupportsNativeShapeShift()).toEqual(false);
+  });
+
+  it("does not support EIP1559", async () => {
+    if (!wallet) return;
+    expect(await wallet.ethSupportsEIP1559()).toEqual(false);
```

Comment:
Is there a reason we aren't supporting EIP1559 with metmask?

Reply:
(@pastaghost probably knows for sure, but IIRC it was just because it wasn't in the MVP AC)

Reply:
@0xean At the time the integration was done, EIP-1559-formatted transactions would sign successfully on the MetaMask extension, but would consistently fail when a Ledger/Trezor was used as the signing agent through MetaMask. Trezor has since released a firmware update to fix this, but I don't know about the status of Ledger. If we can get someone to test this on Ledger, we'd be clear to turn the `ethSupportsEIP1559` flag turned on.